src/app.rs

```diff
@@ -51,12 +53,17 @@ pub use tide::listener::{Listener, ToListener};
 /// constructing an [Api] for each module and calling [App::register_module]. Once all of the
 /// desired modules are registered, the app can be converted into an asynchronous server task using
 /// [App::serve].
+///
+/// Note that the [`App`] is bound to a binary serialization version `VER`. This format only applies
+/// to application-level endpoints like `/version` and `/healthcheck`. The binary format version in
+/// use by any given API module may differ, depending on the supported version of the API.
 #[derive(Debug)]
-pub struct App<State, Error, VER: StaticVersionType> {
+pub struct App<State, Error, VER> {
```

Comment:
It seems like, at this point, you could have gotten rid of the VER param for App entirely...

Reply:
Will do in a follow-up